salt/_modules/metalk8s_network.py

```diff
@@ -0,0 +1,45 @@
+'''Metalk8s network related utilities.'''
+
+import ipaddress
+import logging
+
+from salt.exceptions import CommandExecutionError
+
+log = logging.getLogger(__name__)
+
+__virtualname__ = 'metalk8s_network'
+
+
+def __virtual__():
+    return __virtualname__
+
+
+def pick_nth_ip(service_network, index):
+
+    '''
+    Pick the coredns and kubernetes cluster ip address from service network.
+
+    :param service_network (str): CIDR presentation format read from pillars.
+     For example, '192.168.1.0/24'.
+    :param index (int): Kubernetes cluster IP address is set to 2nd address in
+     service_network.
+     coredns cluster IP address is set to the 10th address in the
+     service network
+
+    :returns (str):
+        An index param of 10 returns => coredns cluster ip
+        An index param of 2 returns => kubernetes cluster ip
+    '''
+    log.info('Picking the nth address from service network')
+
+    network = ipaddress.IPv4Network(unicode(service_network))
+    for i, ip in enumerate(network):
```

Comment:
could be simplified as:

```python
ip = next(itertools.islice(network.hosts(), index, None), None)
if ip is None:
    raise CommandExecutionError(
        'Could not obtain an IP in the network range {}'.format(
            service_network
        )
    )
return str(ip)
```

Reply:
- Adds one more import for `itertools` python module
- `network.hosts()` looks to return strictly the usable set of IP addresses meaning passing an index of 2 returns =>  a CoreDNS IP of 192.168.1.3.
This is not exactly what we want even though we can make a few changes else where to match this.

I prefer to stick to what is currently written. :)

Reply:
I'm not convinced it's a "simplification", I feel it's quite harder to read :/

However, since using `ipaddress` may not be the chosen strategy, it may not make sense to bother further with this...